newsroom/push/publishing.py

```diff
@@ -298,3 +300,27 @@ async def publish_planning_into_event(self, planning: dict[str, Any]) -> Optiona
             # Replace the original document
             await service.update(agenda["_id"], agenda)
             return agenda["_id"]
+
+    async def publish_planning_related_events(self, planning: dict[str, Any], related_event: dict[str, Any]):
+        if not related_event.get("link_type") == "secondary":
+            return None
+
+        service = AgendaItemService()
+        event_id = related_event.get("_id")
+
+        event = await service.find_by_id(event_id)
+        agenda = event.to_dict()
```

Comment:
Instead of using the entire item dict for updates, you can instead construct a list of updates with only the required fields, starting with an empty dict. For example:

```python
from newsroom.types import AgendaItemType

event = await service.find_by_id(event_id)
updates: dict[str, Any] = {}

if event.item_type == AgendaItemType.EVENT and planning and planning.get("guid") not in event.planning_ids:
    updates["planning_ids"] = event.planning_ids + [planning["guid"]]

...

await service.update(event_id, updates)
```

Notice also how above I am using the `event` as a ResourceModel instance and not a dictionary. It's better to use instances of the model instead of a dictionary.